sdks/builds/wasm.mk

```diff
@@ -25,88 +38,120 @@ WASM_RUNTIME_CONFIGURE_FLAGS = \
 	--disable-icall-tables \
 	--with-bitcode=yes
 
-EMSCRIPTEN_VERSION=1.38.11
-EMSCRIPTEN_SDK_DIR=$(TOP)/sdks/builds/toolchains/emsdk
-
-$(TOP)/sdks/builds/toolchains/emsdk:
-	git clone https://github.com/juj/emsdk.git $(EMSCRIPTEN_SDK_DIR)
-
-.stamp-wasm-toolchain: | $(TOP)/sdks/builds/toolchains/emsdk
-	cd $(TOP)/sdks/builds/toolchains/emsdk && ./emsdk install sdk-$(EMSCRIPTEN_VERSION)-64bit
-	cd $(TOP)/sdks/builds/toolchains/emsdk && ./emsdk activate --embedded sdk-$(EMSCRIPTEN_VERSION)-64bit
+.stamp-wasm-runtime-toolchain:
 	touch $@
 
-.stamp-wasm-runtime-toolchain: .stamp-wasm-toolchain
+.stamp-wasm-runtime-$(CONFIGURATION)-configure: $(TOP)/configure $(if $(IGNORE_PROVISION_WASM),,provision-wasm)
+	mkdir -p $(TOP)/sdks/builds/wasm-runtime-$(CONFIGURATION)
+	cd $(TOP)/sdks/builds/wasm-runtime-$(CONFIGURATION) && source $(TOP)/sdks/builds/toolchains/emsdk/emsdk_env.sh && CFLAGS="-Os -g" emconfigure $(TOP)/configure $(WASM_RUNTIME_AC_VARS) $(WASM_RUNTIME_CONFIGURE_FLAGS)
 	touch $@
 
-.stamp-wasm-runtime-configure: $(TOP)/configure
-	mkdir -p $(TOP)/sdks/builds/wasm-runtime
-	cd $(TOP)/sdks/builds/wasm-runtime && source $(TOP)/sdks/builds/toolchains/emsdk/emsdk_env.sh && CFLAGS="-Os -g" emconfigure $(TOP)/configure $(WASM_RUNTIME_AC_VARS) $(WASM_RUNTIME_CONFIGURE_FLAGS)
-	touch $@
+.PHONY: .stamp-wasm-runtime-configure
+.stamp-wasm-runtime-configure: .stamp-wasm-runtime-$(CONFIGURATION)-configure
 
+.PHONY: build-custom-wasm-runtime
 build-custom-wasm-runtime:
-	source $(TOP)/sdks/builds/toolchains/emsdk/emsdk_env.sh && make -C wasm-runtime
+	source $(TOP)/sdks/builds/toolchains/emsdk/emsdk_env.sh && $(MAKE) -C wasm-runtime-$(CONFIGURATION)
+
+.PHONY: setup-custom-wasm-runtime
+setup-custom-wasm-runtime:
+	mkdir -p $(TOP)/sdks/out/wasm-runtime-$(CONFIGURATION)
 
 .PHONY: package-wasm-runtime
 package-wasm-runtime:
-	$(MAKE) -C $(TOP)/sdks/builds/wasm-runtime/mono install
-
-.PHONY: clean-wasm
-clean-wasm:
-	rm -rf .stamp-wasm-runtime-toolchain $(TOP)/sdks/builds/toolchains/emsdk
+	$(MAKE) -C $(TOP)/sdks/builds/wasm-runtime-$(CONFIGURATION)/mono install
 
 .PHONY: clean-wasm-runtime
 clean-wasm-runtime:
-	rm -rf .stamp-wasm-runtime-configure $(TOP)/sdks/builds/wasm-runtime $(TOP)/sdks/builds/wasm-runtime.config.cache $(TOP)/sdks/out/wasm-runtime
+	rm -rf .stamp-wasm-runtime-toolchain .stamp-wasm-runtime-$(CONFIGURATION)-configure $(TOP)/sdks/builds/wasm-runtime-$(CONFIGURATION) $(TOP)/sdks/builds/wasm-runtime-$(CONFIGURATION).config.cache $(TOP)/sdks/out/wasm-runtime-$(CONFIGURATION)
 
 TARGETS += wasm-runtime
 
-UNAME := $(shell uname -s)
```

Comment:
How come we don't need this?

Reply:
It's defined at https://github.com/mono/mono/blob/6d5bf96f1c7f01c223beb9363359a117ae8a505f/sdks/builds/Makefile#L13